libraries/AP_HAL_SITL/sitl_rangefinder.cpp

```diff
@@ -44,23 +43,26 @@ void SITL_State::_update_rangefinder(float range_value)
     }
 
     float voltage = 5.0f;
-    if (fabs(_sitl->state.rollDeg) < 90 &&
-            fabs(_sitl->state.pitchDeg) < 90) {
-        // adjust for apparent altitude with roll
-        altitude /= cosf(radians(_sitl->state.rollDeg)) * cosf(radians(_sitl->state.pitchDeg));
 
-        altitude += _sitl->sonar_noise * _rand_float();
+    if (range_value == -1) {
+        if (fabs(_sitl->state.rollDeg) < 90 &&
+                fabs(_sitl->state.pitchDeg) < 90) {
+            // adjust for apparent altitude with roll
+            altitude /= cosf(radians(_sitl->state.rollDeg)) * cosf(radians(_sitl->state.pitchDeg));
+        }
+    }
 
-        // Altitude in in m, scaler in meters/volt
-        voltage = altitude / _sitl->sonar_scale;
-        voltage = constrain_float(voltage, 0, 5.0f);
+    altitude += _sitl->sonar_noise * _rand_float();
```

Comment:
Ho yes I agreed ... SITL those would lead to incorrect reading ... I was reading the 90degree check just as a check for attitude but it is more . Thank you to correct my everyday errors ...
And it should use is_equal ...